cogs/moderation.py

```diff
@@ -20,91 +129,216 @@ def __init__(self, bot: Bot) -> None:
-        embed.set_thumbnail(url=member.avatar_url)
-        embed.set_author(name=member.name, url=member.avatar_url)
-        embed.set_footer(text=member.guild.name, icon_url=member.guild.icon_url)
+        if not isinstance(member, Member):
+            embed = Embed(
+                title="You can't kick this user",
+                description=textwrap.dedent(
+                    f"""
+                    {member.mention} (`{member.id}`) doesn't seem to be a member of this server.
 
-        await ctx.send(embed=embed)
-        await member.send(embed=embed)
+                    **❯❯ You can only kick server members.**
+                    """
+                ),
+                color=discord.Color.red(),
+            )
+            await ctx.send(f"Sorry {ctx.author.mention}", embed=embed)
+            return
+
+        server_embed = discord.Embed(
+            title="User Kicked",
+            description=textwrap.dedent(
+                f"""
+                **Reason**: {reason}
+                **User**: {member.mention} (`{member.id}`)
+                **Actor**: {ctx.author.mention} (`{ctx.author.id}`)
+                """
+            ),
+            color=discord.Color.orange(),
+            timestamp=datetime.utcnow(),
+        )
+        server_embed.set_thumbnail(url=member.avatar_url_as(format="png", size=256))
+
+        dm_embed = discord.Embed(
+            title="You were Kicked",
+            description=textwrap.dedent(
+                f"""
+                {reason}
+
+                *Server: {ctx.guild.name}*
+                """
+            ),
+            color=discord.Color.red(),
+            timestamp=datetime.utcnow(),
+        )
+        dm_embed.set_thumbnail(url=ctx.guild.icon_url)
+
+        await ctx.send(embed=server_embed)
+        await member.send(embed=dm_embed)
         await member.kick(reason=reason)
 
     @command()
-    @bot_has_permissions(ban_members=True)
     @has_permissions(ban_members=True)
-    async def ban(self, ctx: Context, member: discord.Member, *, reason: str = "No Reason Stated.") -> None:
+    @follow_roles()
+    async def ban(self, ctx: Context, member: UserID, *, reason: str = "No specific reason.") -> None:
         """Ban a User."""
-        embed = discord.Embed(title="Infraction information", color=discord.Color.red())
-        embed.add_field(name="Type", value="Ban")
-        embed.add_field(name="Reason", value=reason)
-        embed.set_thumbnail(url=member.avatar_url)
-        embed.set_author(name=member.name, url=member.avatar_url)
-        embed.set_footer(text=member.guild.name, icon_url=member.guild.icon_url)
+        server_embed = discord.Embed(
+            title="User Banned",
+            description=textwrap.dedent(
+                f"""
+                **Reason**: {reason}
+                **User**: {member.mention} (`{member.id}`)
+                **Actor**: {ctx.author.mention} (`{ctx.author.id}`)
```

Comment:
```suggestion
                **Moderator**: {ctx.author.mention} (`{ctx.author.id}`)
```

Reply:
Fixed in cae3d0229dfe10db6f1ebd9ecdde6bc98c5f61d7